pkg/operator/sync.go

```diff
@@ -34,17 +34,31 @@ func (optr *Operator) syncAll(config *OperatorConfig) error {
 				// return the outer error.
 				glog.Errorf("Error syncing ClusterOperatorStatus: %v", err)
 			}
-			glog.Errorf("Error syncing cluster api controller: %v", err)
+			glog.Errorf("Error syncing machine-api-controller: %v", err)
 			return err
 		}
-		glog.V(3).Info("Synced up all components")
+		glog.V(3).Info("Synced up all machine-api-controller components")
+	}
+
+	// In addition, if the Provider is BareMetal, then bring up
+	// the baremetal-operator pod
+	if config.BaremetalControllers.BaremetalOperator != "" {
```

Comment:
does the baremetal actuator depends on this to run successfully? if so you'd need to ensure this gets deployed before `syncClusterAPIController`

Reply:
The baremetal actuator does not depend on the metal3 pod to run successfully (but needs it to be running to be useful.) It is OK for the baremetal actuator if the metal3 pod is available later. Also, if the metal3 pod deployment wasn't successful we would still want to go ahead with the machine-api-controller pod deployment. For these reasons, would like to keep the ordering as is.